python/python/bystro/proteomics/annotation_interface.py

```diff
@@ -614,29 +616,48 @@ async def execute_query(
-        structs_of_arrays (bool): Whether to return structs of arrays, defaults to True.
-        melt_by_samples (bool): Whether to melt the DataFrame by samples, defaults to False.
-        melt_by_field (str | None): A field to melt by, defaults to None.
-        force_flatten_melt_by_field (bool):
-            When melting by a field, whether to force flatten array values, defaults to False.
-        primary_keys (dict[str, str] | None): A dictionary of primary keys for tracks, defaults to None.
+        fields (list[str] | None):
+            A list of fields to include in the DataFrame.
+        structs_of_arrays (bool)
+             Whether to return structs of arrays, defaults to True.
+        melt_samples (bool):
+            Whether to unpivot `heterozygotes`, `homozygotes`, and `missingGenos` fields.
+            When `True` the resulting DataFrame will have 2 new columns: `samples` and `dosage`,
+            and `heterozygotes`, `homozygotes`, and `missingGenos` will be removed.
+
+            The `dosage` column will have values of 1, 2, or -1, corresponding to whether the sample was
+            found in the `heterozygotes`, `homozygotes`, and `missingGenos` columns, respectively.
+
+            The `samples` column will have the sample ID for each row, and this will always be a
+            scalar value, even if the original `heterozygotes`, `homozygotes`, or `missingGenos` columns
+            had multiple values.
+
+            Defaults to False.
+        explode_field (str | None):
+            A field to explode, converting rows with list values in this column, into
+            multiple rows with 1 value per column, defaults to None.
+        force_flatten_exploded_field (bool):
+            When exploding a field, whether to force flatten array values in cases where the
+            primary key for the track is not present, or the column's value is a list with respect
+            to the primary key. Defaults to False.
+        primary_keys (dict[str, str] | None):
+            A dictionary of primary keys for tracks, defaults to None.
 
     Returns:
         pd.DataFrame: A DataFrame of query results.
     """
     results: list[dict] = []
     search_after = None  # Initialize search_after for pagination
 
-    if melt_by_field is not None:
-        if fields is not None and melt_by_field not in fields:
+    if explode_field is not None:
+        if fields is not None and explode_field not in fields:
             raise ValueError(
-                f"melt_by_field={melt_by_field} is not in fields={fields}, but must be present."
+                f"explode_field={explode_field} is not in fields={fields}, but must be present."
             )
 
         if structs_of_arrays is False:
             raise ValueError(
-                "Cannot yet, melt by field when structs_of_arrays is False, "
+                "Cannot yet, explode field when structs_of_arrays is False, "
```

Comment:
I'm a little confused what this error message means

Reply:
I’ll improve it. It means that you are trying to explode list values in say refseq.name2, but there is no refseq.name2 column, because you’ve chosen the array of structs output, where there is a refSeq column that is a dict of {name2: gene, name: transcript, etc}. 

Say one of those fields is array values. We would need to generate multiple dicts with one value each for that key in each of them. We don’t do that yet.